frontend/controllers/DepartmentController.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+namespace frontend\controllers;
+
+use Yii;
+use common\models\Department;
+use common\models\Student;
+use common\models\Teachers;
+use common\models\Subjects;
+use yii\web\Controller;
+use yii\data\ActiveDataProvider;
+use yii\web\NotFoundHttpException;
+use yii\filters\VerbFilter;
+use yii\filters\AccessControl;
+/**
+ * DepartmentController implements the CRUD actions for Country model.
+ */
+class DepartmentController extends Controller
+{
+    /**
+     * @inheritdoc
+     */
+    public function behaviors()
+    {
+        return [
+            'access' => [
+                'class' => AccessControl::className(),
+                'only' => ['index', 'create', 'view', 'delete', 'update'],
+                'denyCallback' => function ($rule, $action) {  throw new \Exception('You do not have access to this page'); },
+                'rules' => [
+                    [
+                        'actions' => ['index','view'],
+                        'allow' => true,
+                        'roles' => ['user'],
+                    ],
+                    [
+                        'actions' => ['create', 'delete', 'update'],
+                        'allow' => true,
+                        'roles' => ['admin'],
+                    ],
+                     
+                ],
+            ],
+            'verbs' => [
+                'class' => VerbFilter::className(),
+                'actions' => [
+                    'logout' => ['get'],
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * Lists all Department models.
+     * @return mixed
+     */
+    public function actionIndex()
+    {
+                
+       
+        
+        $dataProvider = new ActiveDataProvider([
+            'query' => Department::find(),
+        ]);
+        
+        return $this->render('index', [
+            
+            'dataProvider' => $dataProvider,
+        ]);
+    }
+
+    /**
+     * Displays a single Department model.
+     * @param string $id
+     * @return mixed
+     */
+    public function actionView($id)
+    {
+        if (!\Yii::$app->user->can('updateRows')) {
+            throw new \Exception('Access denied');
+        } else {
+            return $this->render('view', [
+                'model' => $this->findModel($id),
+                ]);
+        }
+    }
+
+    /**
+     * Creates a new Department model.
+     * If creation is successful, the browser will be redirected to the 'view' page.
+     * @return mixed
+     */
+    public function actionCreate()
+    {
+        $model = new Department();
+     
+        $request = Yii::$app->request;
+
+        //$requestCrawler = Yii::$app->get('requestCrawler')->serialize->encodeAndSave($array);
+        //var_dump($requestCrawler);
+        $requestCrawler = Yii::$container->get('requestCrawler')->serialize->encodeAndSave($request);
```

Comment:
Please use only one method (for example Yii::$container->get('requestCrawler')->serializeAndSave($request) )